### PR TITLE
Associations

### PR DESCRIPTION
**Pull request by samuelmoiwa**
## Solution description

In this branch, I created the remaining classes for the school library and create associations between them.

> Created person class
> Created student class
> Created teacher class
> Add decorator classes

## Address affected and ensured
* books.rb
* rental.rb
* student.rb
* classroom.rd
---

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+require_relative './rental'
+
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
+
```
Comment (omar-muhamad) [inline book.rb]: - [x] Please make sure that Getters and Setters for `@rentals` should be present.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @samuelmoiwa,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

Please add the following in `person.rb` to address the relationship between `person` and `rental` (A person has many rentals):
- [x] Getters and Setters for `@rentals` should be present.
- [x] `@rentals` should be initialized in the `initialize` method.
- [x] A method for adding rentals should be present.
https://github.com/samuelmoiwa/OOP-school-library/blob/ee0e281c64c205c4e7031f997c6cfbebe6aca9d5/person.rb#L1-L32


- [x] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: book.rb (hunk 1 above)
Comment (omar-muhamad) [reply book.rb]: - [ ] As I mentioned before, Please add Getters and Setters `attr_accessor` for `@rentals`, not Getters only `attr_reader` as you did.

File: person.rb [hunk 2]
```diff
@@ -9,10 +9,11 @@ def initialize(age, name = 'Unknown', parent_permission: true)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
   end
 
   attr_accessor :name, :age
-  attr_reader :id
+  attr_reader :id, :rentals
```
Comment (omar-muhamad) [inline person.rb]: - [ ] As I mentioned before, Please add Getters and Setters `attr_accessor` for `@rentals`, not Getters only `attr_reader` as you did.
Other than that you have done a great job.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @samuelmoiwa,

Nice job making the required changes. Unfortunately, you missed some of them but you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow. ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by tresorsawasawa — APPROVED** [review]
# STATUS: APPROVE :heavy_check_mark: :heavy_check_mark:  :champagne: 

Hi @samuelmoiwa 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Your README file is professional :heavy_check_mark: 

## Optional suggestions
- [ **OPTIONAL** ] **Kindly make sure you remove **template logo** from your README file as required for each Microverse project. Optionally, you can replace with their own logo if you have one.**
- [ **OPTIONAL** ] Kindly make sure you write meaningful and descriptive messages for your commit for your next projects for best practices.  
_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**